drivers/charger/Kconfig

```diff
@@ -0,0 +1,18 @@
+# Copyright 2023 Cirrus Logic, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig CHARGER
+	bool "Battery charger drivers"
+	help
+	  Enable battery charger driver configuration.
+
+if CHARGER
+
+module = CHARGER
+module-str = charger
+source "subsys/logging/Kconfig.template.log_config"
+
```

Comment:
We can add the init priority here. See https://github.com/zephyrproject-rtos/zephyr/pull/55887 as an example.

Reply:
Ack